### PR TITLE
[Postgresql] Added initialization parts to fuzzers

### PR DESCRIPTION
**Pull request by yooyoo9**
@spq @sfrost 

- Fixed initialization part of json_parser and simple_query_fuzzer

- Set up database to run the queries

- Included dictionary and seed corpus for better coverage

### CODE REVIEW

File: projects/postgresql/build.sh [hunk 1]
```diff
@@ -14,12 +14,24 @@
...
 cp -r $SRC/fuzzer src/backend/
-mkdir bld
+
+useradd fuzzuser
+chown -R fuzzuser .
 cd bld
 
-../configure --prefix /usr/local/
-make -j$(nproc)
+CC="" CXX="" CFLAGS="" CXXFLAGS="" su fuzzuser -c ../configure
```
Comment (Google-Autofuzz) [inline projects/postgresql/build.sh:23]: Why are you defining those variables? There should be empty by default.

File: projects/postgresql/build.sh [hunk 2]
```diff
@@ -14,12 +14,24 @@
 # limitations under the License.
 #
 ################################################################################
-git apply ../add_fuzzers.diff
 cp -r $SRC/fuzzer src/backend/
-mkdir bld
+
+useradd fuzzuser
```
Comment (Google-Autofuzz) [inline projects/postgresql/build.sh:19]: Isn't it ok to run this under the current user?

**Review by Google-Autofuzz — CHANGES_REQUESTED** [review]

File: projects/postgresql/build.sh (hunk 1 above)
Comment (yooyoo9) [reply projects/postgresql/build.sh:23]: When running infra/helpyer.py build_fuzzers, those flags are set to specific values.

File: projects/postgresql/build.sh (hunk 2 above)
Comment (yooyoo9) [reply projects/postgresql/build.sh:19]: Postgresql requires a non-root user to create a database

File: projects/postgresql/build.sh (hunk 1 above)
Comment (jonathanmetzman) [reply projects/postgresql/build.sh:23]: They are set, but I'm surprised they aren't needed here. I guess CI passing https://github.com/google/oss-fuzz/pull/4357/checks?check_run_id=1026019910 means it works though.

**Comment by jonathanmetzman** [comment]
Let me know if this is ready to merge.

File: projects/postgresql/project.yaml [hunk 3]
```diff
@@ -9,4 +9,3 @@ fuzzing_engines:
   - honggfuzz
```
Comment (jonathanmetzman) [inline projects/postgresql/project.yaml:9]: nit: can you drop the space after `auto_ccs` and before the colon please.

**Comment by yooyoo9** [comment]
@jonathanmetzman My co-host reviewed. This is ready to merge.

**Pull request merged**